File: library/src/main/java/com/goldenpie/devs/pincodeview/SinglePinView.java
package com.goldenpie.devs.pincodeview;

import android.content.Context;
import android.os.Build;
import android.support.annotation.Nullable;
import android.support.annotation.RequiresApi;
import android.support.v7.widget.AppCompatImageView;
import android.util.AttributeSet;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.LinearLayout;

/**
 * Created by antoxa2584 on 07.06.17.
 */

class SinglePinView extends LinearLayout {
    private AppCompatImageView outerPinView;
    private AppCompatImageView innerPinView;

    public SinglePinView(Context context) {
        super(context);
        initView();
    }

    public SinglePinView(Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
        initView();
    }

    public SinglePinView(Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        initView();
    }

    @RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
    public SinglePinView(Context context, AttributeSet attrs, int defStyleAttr, int defStyleRes) {
        super(context, attrs, defStyleAttr, defStyleRes);
        initView();
    }

    private void initView() {
        LayoutInflater inflater;
        inflater = (LayoutInflater) getContext()
                .getSystemService(Context.LAYOUT_INFLATER_SERVICE);

        View mView = inflater.inflate(R.layout.single_pin_view_layout, this);

        outerPinView = (AppCompatImageView) mView.findViewById(R.id.outer_pin_view);
        innerPinView = (AppCompatImageView) mView.findViewById(R.id.inner_pin_view);

        outerPinView.setImageResource(R.drawable.circle);
        innerPinView.setImageResource(R.drawable.circle);
    }

    public void setColors(int innerColor, int outerColor) {
        outerPinView.setColorFilter(outerColor);
        innerPinView.setColorFilter(innerColor);
    }

    public AppCompatImageView getOuterPinView() {
        return outerPinView;
    }

    public AppCompatImageView getInnerPinView() {
        return innerPinView;
    }
}

File: library/src/main/java/com/goldenpie/devs/pincodeview/SinglePinView.kt
package com.goldenpie.devs.pincodeview

import android.content.Context
import android.os.Build
import android.support.annotation.RequiresApi
import android.support.v7.widget.AppCompatImageView
import android.util.AttributeSet
import android.view.LayoutInflater
import android.view.View
import android.widget.LinearLayout

internal class SinglePinView : LinearLayout {
    lateinit var outerPinView: AppCompatImageView
        private set
    lateinit var innerPinView: AppCompatImageView
        private set

    constructor(context: Context) : super(context) {
        initView()
    }

    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs) {
        initView()
    }

    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {
        initView()
    }

    @Suppress("unused")
    @RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int, defStyleRes: Int) : super(context, attrs, defStyleAttr, defStyleRes) {
        initView()
    }

    private fun initView() {
        val inflater: LayoutInflater = context
                .getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater

        with(inflater.inflate(R.layout.single_pin_view_layout, this)) {
            outerPinView = findViewById<View>(R.id.outer_pin_view) as AppCompatImageView
            innerPinView = findViewById<View>(R.id.inner_pin_view) as AppCompatImageView
        }
    }

    fun setColors(innerColor: Int, outerColor: Int) {
        outerPinView.setColorFilter(outerColor)
        innerPinView.setColorFilter(innerColor)
    }
}